src/rpm/headers/header.rs

```diff
@@ -612,34 +651,50 @@ impl IndexHeader {
 }
 
 /// A single entry within the [`IndexHeader`](self::IndexHeader)
-#[derive(Debug, PartialEq)]
+#[derive(PartialEq)]
 pub(crate) struct IndexEntry<T: num::FromPrimitive> {
-    pub(crate) tag: T,
+    pub(crate) tag: u32,
     pub(crate) data: IndexData,
     pub(crate) offset: i32,
     pub(crate) num_items: u32,
+
+    // Marks what type of IndexEntry it is
+    entry_type: PhantomData<T>,
+}
+
+/// Custom Debug impl for the benefit of showing the tag name, if we are familiar with it
+impl<T: Tag> std::fmt::Debug for IndexEntry<T> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let tag = T::from_u32(self.tag);
+        let tag_val = if let Some(val) = tag {
```

Comment:
This code feels a bit magical to me, maybe I'm just a bit removed from it. I _think_ because `IndexTag` derives FromPrimitive that the `from_u32` code will actually give us something useful. Not a problem cause it compiles, it's just more opaque than I expected for Rust. I had to load it into an editor to chase down the traits.

Reply:
That's correct.  Want me to call the trait explicitly as was done here? https://github.com/rpm-rs/rpm/pull/137/files#diff-e8b11ac65b082c9412d618eb40c89f86b44fada22b3dbc251ad1d289d72770fbL629